contracts/contracts/strategies/ConvexOUSDMetaStrategy.sol

```diff
@@ -148,12 +143,19 @@ contract ConvexOUSDMetaStrategy is BaseConvexMetaStrategy {
             true
         );
 
-        // always withdraw all of the available metapool LP tokens (similar to how we always deposit all)
-        // slither-disable-next-line unused-return
-        metapool.remove_liquidity(lpToBurn, [uint256(0), uint256(0)]);
-        IVault(vaultAddress).burnForStrategy(
-            metapoolMainToken.balanceOf(address(this))
+        // calculate the min amount of OUSD expected for the specified amount of LP tokens
+        // slither-disable-next-line divide-before-multiply
+        uint256 minOUSDAmount = lpToBurn *
```

Comment:
This line really doesn't feel right to me.

I assume it's supposed to work like:

`min dollar value of pool lp tokens` -`3pool dollar value withdrawn` - 1 < `ousd dollar value withdrawn`

However, the `lpToBurn * metapool.get_virtual_price()` is multiplying two 1e18 numbers together without division. This should mean that this results in a crazy high number, which I would expect would break withdraws due to not meeting the minimum.

Secondly, the `- num3CrvTokens` line feels like it's trying to determine the dollar value of the 3pool LP tokens, but since it's not adjusted for the virtual price, this would make the subtraction side too small, which should make it break with false alarms when the exchange rates are close to equal.

Given that it looks like there are two reasons that this code would not work, I wonder if I'm wrong or if the fork tests are.

Stepping back, it looks like this code is just trying to duplicate what the curve calculations are doing. Is this code gaining us anything other than validating curve math? If not, I would lean toward switching back to using a zero for the OUSD minimum.

Reply:
I've confirmed with fork testing that withdraws are broken. If I changed it to `uint256 minOUSDAmount = 0`, it can withdraw funds.

Reply:
<img width="962" alt="image" src="https://user-images.githubusercontent.com/837/197772010-967808a7-bb46-4402-bb4b-ae21e8062d14.png">